lab1/app.js

```diff
@@ -193,6 +181,105 @@ class StatusTable {
     }
 }
 
+class Inventory {
+    constructor(root, store) {
+        this.root = root;
+        this.store = store;
+		this.foods = this.store.foods;
+        this.init();
+    }
+
+    init () {
+        this.render();
+        // TODO: attach event listeners like click to remove items after rendering
+    }
+
+    destroy () {
+        // TODO: remove event listeners added from the init above
+    }
+
+    removeItem (itemName) {
+        // TODO: function to remove item given item name from store
+		
+		
+    }
+
+    render () {
+        // using innerHTML to render inventory listing
+		//let inventoryTable = this.root.querySelector("tbody");
+		//console.log(this.foods);
+		//inventoryTable.innerHTML = function(this.foods){
+		//	let foods = this.foods;
+		//	let result = " ";
+		//	(var i = 0; i< foods.length; i++){
+		//	result += "<tr><td>"+ food[i].date + "</td><td> " + food[i].name + '</td><td><button class="delete-button" data-id="0">Delete</button></td></tr>';
+		//}
+		//return result;
+	//	}
+    }
+}
+
+class Menu {
+    constructor(root, store) {
+        this.root = root;
+        this.store = store;
+        this.init();
+    }
+
+    init () {
+        this.render();
+    }
+
+    render () {
+        // TODO: render a list of food menu from store using innerHTML
+		
+//		this.store.foods, should support/replace menu
+
+		console.log(this.store.foods);
+		//let result = " ";
+		//	for(var i = 0; i<list.length; i++){
+		//		result += "<tr><td>"+ list[i].name + "</td><td> " + list[i].price + '</td><td><button class="delete-button" data-id="0">Delete</button></td></tr>';
+		//	}
+		//return result;
+//    }
+	}
+}
+
+class CreateFood {
+    constructor(root, store) {
+        this.root = root; // root should be the container of the form itself
+        this.store = store;
+        this.init();
+    }
+
+    init () {
+        // attach click event to create button
+    }
+
+    createFood () {
+        // will need to do querySelector to find out every single form element
+        // to get their values before creating a new food
+        // after creating a new food item, add it to store
+		
+		 let foods = this.store.foods || [];
+		 
+		 let foodItem = this.root.querySelector('.newFoods');
+		
+		console.log(foodItem);
+		//  console.log(foods);
+		//foods.push({ 
+			//		name: foodItem.name,
+				//	img: foodItem.image,
+				//description: foodItem.description,
+						// ingredients: this.store.ingredient ||[]
+		//});
```

Comment:
A starting point for the create food method would be:

```js
let name = document.querySelector('#food_name_input').value;
let description = document.querySelector('#food_description_input').value;
```

And then you can do the push method after.